Controller/Annotations/AbstractScalarParam.php

```diff
@@ -49,10 +51,17 @@ public function getConstraints()
                 ),
             ));
         } elseif (is_array($this->requirements) && isset($this->requirements['rule']) && $this->requirements['error_message']) {
+            @trigger_error('Using the "rule" and "error_message" options as requirements is deprecated. Use an array of constraints instead.', E_USER_DEPRECATED);
```

Comment:
Sorry to bother, rethinking about it, I think in https://github.com/FriendsOfSymfony/FOSRestBundle/pull/1752/files#r136455286 I meant deprecating using arrays not only containing `Constraint`s, scalar requirements seems like a nice syntactic sugar to me.